### ISSUE TITLE
`require-presentational-children` fails for named blocks

### ISSUE THREAD

**Issue by joukevandermaas** [ember-template-lint/ember-template-lint#2654]
We have a button component that yields several named blocks for various reasons (mostly used by other re-usable components). The rule `require-presentational-children` fails whenever we add e.g. `role="tab"` to one of these buttons:

```
<MyButton role="tab">
  <:default>Button text</:default>
</MyButton>
```

Fails with the error

> <MyButton> has a role of tab, it cannot have semantic descendants like <:default>

I think named blocks should not be flagged like this, rather the contents should be evaluated instead (if anything). The result is now that people are encouraged to *remove* the role attribute or suppress the lint rule, as those are the only ways to fix the error.
